rules_java_gapic/java_gapic_spring.bzl

```diff
@@ -0,0 +1,135 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@rules_gapic//:gapic.bzl", "proto_custom_library")
+
+def _java_gapic_spring_postprocess_srcjar_impl(ctx):
+    gapic_srcjar = ctx.file.gapic_srcjar
+    output_srcjar_name = ctx.label.name
+    srcjar_name = output_srcjar_name + "_raw.srcjar"
+    output_spring = ctx.outputs.spring
+    formatter = ctx.executable.formatter
+
+    output_dir_name = ctx.label.name
+    output_dir_path = "%s/%s" % (output_spring.dirname, output_dir_name)
+
+    script = """
+    WORKING_DIR=`pwd`
+    cd $WORKING_DIR
+    mkdir {output_dir_path}/spring
+    unzip -q {gapic_srcjar}
+    unzip -q temp-codegen-spring.srcjar -d {output_dir_path}
+    # This may fail if there are spaces and/or too many files (exceed max length of command length).
+    {formatter} --replace $(find {output_dir_path}/spring -type f -printf "%p ")
+
+    # Spring source files.
+    cd {output_dir_path}/src/main/java
+    zip -r $WORKING_DIR/{output_srcjar_name}.srcjar ./
+
+    cd $WORKING_DIR
+
+    mv $WORKING_DIR/{output_srcjar_name}.srcjar {output_spring}
+    """.format(
+        output_srcjar_name = output_srcjar_name,
+        gapic_srcjar = gapic_srcjar.path,
+        srcjar_name = srcjar_name,
+        formatter = formatter,
+        output_dir_name = output_dir_name,
+        output_dir_path = output_dir_path,
+        output_spring = output_spring.path,
+    )
+
+    ctx.actions.run_shell(
+        inputs = [gapic_srcjar],
+        tools = [formatter],
+        command = script,
+        outputs = [output_spring],
+    )
+
+_java_gapic_spring_postprocess_srcjar = rule(
+    attrs = {
+        "gapic_srcjar": attr.label(mandatory = True, allow_single_file = True),
+        "formatter": attr.label(
+            default = Label("//:google_java_format_binary"),
+            executable = True,
+            cfg = "host",
+        ),
+    },
+    outputs = {
+        "spring": "%{name}-spring.srcjar",
+    },
+    implementation = _java_gapic_spring_postprocess_srcjar_impl,
+)
+
+def java_gapic_spring_library(
+        name,
+        srcs,
+        grpc_service_config = None,
+        gapic_yaml = None,
+        service_yaml = None,
+        **kwargs):
```

Comment:
I suppose this is derived from [`_java_gapic_library()`](https://github.com/googleapis/gapic-generator-java/blob/main/rules_java_gapic/java_gapic.bzl#L246) in `java_gapic.bzl` and you removed a couple of options.  Can you double check if they don't affect sping-codegen for us?
E.g. I noticed `transport` is removed, If I get it right, this is where transport is specified as `grpc` or `grpc+rest` or `rest`, which get parsed in and eventually plays a part in #1078. Are we losing this info here?

Reply:
Oh, we indeed need `transport` to be specified! I'll double check the impact of the other options.

Reply:
So, the other option is `rest_numeric_enums`, which is used by `HttpJsonServiceStubClassComposer` to determine how to serialize enums (by number or value). I see that our spring composers do not mention enums, so it should be safe to remove.

Reply:
It is probably not in scope now, but once SpringCodeGen decide to support both `grpc` and `rest`, we need a way to pass the correct value to this Spring specific bazel rule, and currently the only source of truth is the Bazel files in googleapis.